activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java

```diff
@@ -261,16 +254,9 @@ public Task update(UpdateTaskPayload updateTaskPayload) {
             internalTask.setFormKey(updateTaskPayload.getFormKey());
         }
         //@TODO: add check to see if something was changed before saving + add all other updateable values
-        taskService.saveTask(internalTask);
-        Task convertedTask = task(updateTaskPayload.getTaskId());
-        fireTaskUpdatedEvent(convertedTask);
-        return convertedTask;
-    }
-
-    private void fireTaskUpdatedEvent(Task convertedTask) {
-        for (TaskRuntimeEventListener<TaskUpdatedEvent> listener : taskUpdatedListeners) {
-            listener.onEvent(new TaskUpdatedEventImpl(convertedTask));
-        }
+        org.activiti.engine.task.Task updatedTask = taskService.saveTask(internalTask);
```

Comment:
Here, I simplified the code and modified taskService.saveTask to return updated task entity without need to requery task to return the result.